### ISSUE TITLE
failed to open /dev/urandom: Too many open files

### ISSUE THREAD

**Issue by garyyu** [GrinPlusPlus/GrinPlusPlus#59]
```
$ ulimit -n
256
```

Crash log:
```
libc++abi.dylib: terminating with uncaught exception of type std::__1::system_error: random_device failed to open /dev/urandom: Too many open files
Abort trap: 6
```

The workaround is to increase the ulimit, but we need investigate whether we have correct releasing for `/dev/urandom`.

**Comment by garyyu** [comment]
This one is very annoying, even set `ulimit -n 512`, still frequently crash for this `Too many open files`.

**Comment by DavidBurkett** [comment]
Yea, you need way more than that. I set it to 4096, I believe, but I don't know what the lowest we can go is.

**Comment by DavidBurkett** [comment]
RocksDB is the cause. I think I have a workaround. You can set options.max_open_files when opening each RocksDB instance. I will push the changes after I get a chance to test some more and find a reasonable value for max_open_files.

**Comment by DavidBurkett** [comment]
Should be fixed in v0.6.2. Feel free to reopen if you have the issue again
